sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -233,6 +218,49 @@ public String promptForMissingValueWithOptions(String message, String value, Str
 		return promptForMissingValueWithOptions(message, value, parameterName, options, null, null);
 	}
 
+	@Override
+	public String promptForArtifactVersion(String message, Artifact artifact, String otherMessage, VersionsHelper versionsHelper) throws MojoExecutionException  {
+		List<String> allVersions = versionsHelper.getSuggestedVersions(artifact, null);
+		if (SDKConstants.REFAPP_2X_GROUP_ID.equals(artifact.getGroupId()) && SDKConstants.REFAPP_2X_ARTIFACT_ID.equals(artifact.getArtifactId())) {
+			for (String version : SDKConstants.SUPPPORTED_REFAPP_VERSIONS_2_3_1_OR_LOWER) {
+				if (!allVersions.contains(version)) {
+					allVersions.add(version);
+				}
+			}
+		}
+		return promptForArtifactVersion(message, allVersions, 0, MAX_OPTIONS_SIZE, otherMessage);
+	}
+
+	protected String promptForArtifactVersion(String message, List<String> options, int pageNum, int pageSize, String otherMessage) throws MojoExecutionException {
+		int fromIndex = pageNum * pageSize;
+		int toIndex = fromIndex + pageSize - 1;
+		List<String> currentPage = new ArrayList<>();
+		for (int i = fromIndex; i <= toIndex && i < options.size(); i++) {
+			currentPage.add(options.get(i));
+		}
+		boolean isFirstPage = pageNum == 0;
+		boolean isLastPage = (pageNum + 1) * pageSize >= options.size();
+
+		if (!isFirstPage) {
+			currentPage.add(HIGHER_VERSIONS);
+		}
+		if (!isLastPage) {
+			currentPage.add(LOWER_VERSIONS);
+		}
+
+		String otherPrompt = (isLastPage ? otherMessage : null);
+		String answer = promptForMissingValueWithOptions(message, null, null, currentPage, otherPrompt, null);
+
+		if (HIGHER_VERSIONS.equals(answer)) {
+			return promptForArtifactVersion(message, options, pageNum - 1, pageSize, otherMessage);
+		}
+		else if (LOWER_VERSIONS.equals(answer)) {
+			return promptForArtifactVersion(message, options, pageNum + 1, pageSize, otherMessage);
+		}
+
+		return answer;
+	}
```

Comment:
This is the main code that now drives the Wizard when one is selecting a platform, 2.x refapp or 3.x refapp version.  This downloads all suggested versions and allows paging through all of them, rather than just presenting the most recent 5 or 6 or whatever.